test/upload-download.test.js

```diff
@@ -91,6 +91,7 @@ describe('storage service', function() {
       .set('Content-Type', 'application/json')
       .expect('Content-Type', /json/)
       .expect(200, function(err, res) {
+        if (err) done(err);
```

Comment:
>is there a difference whether its returned?
>the callback ends the test
You call the callback, but that does not mean it terminates the execution (unless you return the callback) it also goes to the next line which is `verifyMetadata(res.body, 'test-container');` in your case. @pthieu @bajtos to confirm.